client/src/components/board-details/SidebarButton.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import { usePathname } from "next/navigation";
+import Link from "next/link";
+import { ReactNode } from "react";
+import { cn } from "@/lib/utils";
+import { Button } from "@/components/ui/button";
+
+interface SidebarButtonProps {
+  href: string;
+  children: ReactNode;
+}
+
+export const SidebarButton = ({ href, children }: SidebarButtonProps) => {
+  const pathname = usePathname();
+  const isActive = () => {
+    const normalizedPath = pathname.endsWith("/") ? pathname.slice(0, -1) : pathname;
+    const normalizedHref = href.endsWith("/") ? href.slice(0, -1) : href;
+
+    return normalizedPath === normalizedHref;
+  };
```

Comment:
I'm not sure if the `SidebarButton` component itself should determine if it's active or not. Don't you think that passing down a prop `isActive: boolean` would be a better approach?

Reply:
Okay done